### PR TITLE
nRF5340 multiprotocol fixes

### PR DESCRIPTION
**Pull request by jciupis**
The cherry-picked commits originate from https://github.com/zephyrproject-rtos/zephyr/pull/31474

### CODE REVIEW

**Review by czeslawmakarski — APPROVED** [review]

**Review by e-rk — APPROVED** [review]

**Review by rlubos — COMMENTED** [review]
Looks good, but we need to wait for the final SHA in the manifest update anyway.

**Comment by mbolivar-nordic** [comment]
Please rebase this PR if still relevant; master has been rebased.

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 1]
```diff
@@ -104,7 +104,7 @@ static void nrf5_get_eui64(uint8_t *mac)
 				   sizeof(factoryAddress));
 #endif
 	if (ret != 0) {
-		LOG_ERR("Unable to read EUI64 from the secure zone.")
+		LOG_ERR("Unable to read EUI64 from the secure zone.");
```
Comment (mbolivar-nordic) [inline drivers/ieee802154/ieee802154_nrf5.c:107]: Please add a note in the commit log saying this should be squashed into 0ab7df74b22.

**Review by mbolivar-nordic — APPROVED** [review]
Looks good pending west.yml change to SHA. Please update the commit log on the noup match before merging as well.

**Comment by jciupis** [comment]
This PR was yielded obsolete by #445.

**Pull request closed without merging**